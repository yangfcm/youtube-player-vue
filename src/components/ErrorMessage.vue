<script setup lang="ts">
defineProps<{
  message?: string
}>()
</script>

<template>
  <div>
    <v-alert icon="mdi-alert" type="error">{{ message || 'Oops, something went wrong.' }}</v-alert>
  </div>
</template>
