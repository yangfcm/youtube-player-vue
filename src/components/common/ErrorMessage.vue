<template>
  <div>
    <app-blank></app-blank>
    <h3 class="ui red header center aligned">
      <i class="exclamation triangle icon"> </i><slot></slot>
    </h3>
    <app-blank></app-blank>
  </div>
</template>

<script>
import Blank from "./Blank";
export default {
  components: {
    appBlank: Blank,
  },
  props: ["errorMessage"],
};
</script>
<style scoped></style>
