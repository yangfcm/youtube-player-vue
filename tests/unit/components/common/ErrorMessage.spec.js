import { shallowMount } from "@vue/test-utils";
import ErrorMessage from "@/components/common/ErrorMessage";

test("ErrorMessage component can render correctly", () => {
  const errorMessage = "Error occurs!";
  const wrapper = shallowMount(ErrorMessage, {
    slots: {
      default: errorMessage,
    },
  });
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("h3").text()).toContain(errorMessage);
});
