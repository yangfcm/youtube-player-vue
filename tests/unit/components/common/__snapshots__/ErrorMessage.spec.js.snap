// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component can render correctly 1`] = `
<div>
  <app-blank-stub></app-blank-stub>
  <h3 class="ui red header center aligned"><i class="exclamation triangle icon"></i>Error occurs!</h3>
  <app-blank-stub></app-blank-stub>
</div>
`;
